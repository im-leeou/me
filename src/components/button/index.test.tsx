import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Button } from './index';

describe('components-button', () => {
  it('renders correctly', () => {
    const result = render(<Button />);

    expect(result).toMatchSnapshot();
  });
});
