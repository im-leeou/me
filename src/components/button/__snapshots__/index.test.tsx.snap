// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components button > renders as full width 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button > renders with bigger icon on bigger screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Matched label
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button > renders with both icons 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Matched label
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button > renders with end icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Matched label
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components button > renders with start icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus w-full"
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Matched label
      </button>
    </div>
    <div>
      <button
        class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="center focus-shadow text-sm-rps ring-border transition-300 rounded-2 bg-black-500 px-4 py-2.5 text-white-500 outline-none ring-black-400 hover:bg-black-700 hover:ring-black-500 focus:shadow-focus"
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
